tracer/src/Datadog.Trace/IAST/Evidence.cs

```diff
@@ -18,4 +18,9 @@ public Evidence(string value, Range[]? ranges = null)
     public string Value { get; }
 
     public Range[]? Ranges { get; }
+
+    public override int GetHashCode()
```

Comment:
Would using` readonly record struct `make sense here, you would get `gethashcode` and equality for free. Same question for the other structs
Edit: I realize you actually only use the hashcode so don't mind me

Reply:
It was as a readonly struct for performance reasons. In this particular case, the gethashcode override would be needed anyway to take into account the array contents.